### ISSUE TITLE
Include sass version

### ISSUE THREAD

**Issue by vuhrmeister** [Zodiase/meteor-mdl#16]
I would like to be able to use mixins and varialbes defined in mdl project. So it would be nice if they were available.
Otherwise I need to include and maintain them separately.

**Comment by SPAHI4** [comment]
+

**Comment by Zodiase** [comment]
I'll see what I can do.

**Comment by pdxbmw** [comment]
At this point, this is a big limitation with using MDL and Meteor; there doesn't seem to be an _easy_ way to access SASS resources. I'm using your `meteor-mdl` library, but have had to manually download the SASS stylesheets from `getmdl.io` and store them in my client-side assets folder. Further, I had to modify the stylesheet `import` statement paths for it to function properly, which was cumbersome and problematic. Having direct access to the SASS files from your package would be an incredibly helpful feature.

For what it's worth, your package is the only implementation of MDL in Meteor that worked correctly for me. Kudos.

**Comment by pdxbmw** [comment]
I believe I have a solution for this. I'm a bit tired at the moment so I'll look it over in the morning and create a pull request if it all checks out.

**Comment by Zodiase** [comment]
@vuhrmeister  I'm not clear about something; if I include a sass file in the package, would you be able to directly use it? Because I think in the end meteor would process all package files into final css and js files, at which point there are no sass files whatsoever. Am I missing something here?

**Comment by RobinVanCauter** [comment]
We actually pulled your package and integrated a SCSS version of it, been using it for quite a while now without any issues.

I uploaded it so you can have a look at it: https://github.com/moonkinco/moonkin-mdl

Might be a bit messy in some places because we did not actually intend on sharing it anytime soon (we don't have time to actually properly maintain a package like that at this point) and it's probably better if you just implement it in your package since it's pretty much a fork of your package anyway :D

This is how we use it in our main scss file:

```
// Mixins
@import '{moonkin:mdl}/mixins';

// Resets and dependencies
@import '{moonkin:mdl}/resets/resets';
@import '{moonkin:mdl}/typography/typography';

// Components
@import '{moonkin:mdl}/palette/palette';
@import '{moonkin:mdl}/ripple/ripple';
@import '{moonkin:mdl}/animation/animation';
@import '{moonkin:mdl}/badge/badge';
@import '{moonkin:mdl}/button/button';
...
```

**Comment by ramijarrar** [comment]
@Zodiase Any updates on this?

**Comment by Zodiase** [comment]
@T-Grave It that the only way to allow scss files to be accessible to the app? If I don't use the `fourseven:scss` package and the `{isImport:true}` option, would that still work? I'm asking because I'm worried that these scss files would be converted to css code and sent to the client which would cause issues.

**Comment by Zodiase** [comment]
Also I'd like to ask: if you have access to those raw scss files (directly from google's mdl), would you still need this package to add the built css file? Note that during the build process some images were embedded into the css code so if you are just using those scss files, you are missing those images and could have issues with that.

**Comment by ramijarrar** [comment]
You can fix the images problem by serving them like you have done with the fonts. 

I have a functioning example here: https://github.com/talos-code/mdl

**Comment by Zodiase** [comment]
@ramijarrar That's not going to work nicely though. I haven't tested but I'm guessing in your example it might not work at all due to the incorrect path. Let's say the meteor app is running at `/app/`, then the files added by packages would be accessible at `/app/packages/package_name/file.ext`. <del>If the scss files were built into css code and served by this package then the css file is likely located at `/app/packages/package_name/css.css` while the images are located at `/app/packages/package_name/images/svg.svg` which is fine</del> (See edits below). The problem is, if that's not the case and the users want to use the scss files and build their own css file (which I believe is more or less this issue is about), then the css file is going to be at `/app/css.css` and thus it won't be able to reach `/app/packages/package_name/images/svg.svg` with `url(images/svg.svg)`, which is bad. In the case when other packages want to take advantages of the scss files then that wouldn't work either. So those image files really are meant to be embedded.

Edit:
Actually I just tested and found out that in development mode meteor concatenates all css into one file. So the css code from the packages would be at `/app/merged-stylesheets.css` and it will never reach `/app/packages/package_name/images/svg.svg` with `url(images/svg.svg)`.

**Comment by ramijarrar** [comment]
Notice that I am correcting this by setting the $image_path variable

**Comment by Zodiase** [comment]
@ramijarrar Oh sorry didn't spot that. Although the path should be `packages/talos_mdl/images` (without the leading "/") or it doesn't work with apps not running at root.

Now it looks like it should work. I'll setup a branch for testing soon.

**Comment by Zodiase** [comment]
The test branch is up with the above commit. Check it out and if it's good then I'll publish it ASAP as `1.0.6_3`.

Edit:
I'm still using the pre-built css file and these added scss files are only for importing. Issues related to themes and color settings please refer to issue #7. Though this branch also includes changes proposed in that issue.

**Comment by ramijarrar** [comment]
How will this cater for people who don't want to use Sass?

**Comment by ramijarrar** [comment]
Also, how are you handling the issue with the images here? Is the user expected to change the variable himself?

**Comment by ramijarrar** [comment]
So will this mean essentially loading mdl twice when importing the Sass?

**Comment by Zodiase** [comment]
@ramijarrar This branch is specifically for the PO who wants to "use mixins and varialbes defined in mdl project". I'm guessing some users may want to come up with other components/designs that follow the styles of mdl so it could be helpful to reuse. If they don't, then this change doesn't make any difference since the added files are flagged with `{isImport: true}` and will not be added to the merged css file. This also means I don't need to take care of the path issue as nothing is broken by this change.

**Comment by ramijarrar** [comment]
So in that branch the pre-built css is not included?

**Comment by Zodiase** [comment]
@ramijarrar It is included. However the scss files will not affect the merged css file due to that import flag. I expect whoever wants to use the scss files should know the consequences. For example, if they import a scss file of the checkbox, then sure that part of the css code would essentially be sent twice, and of course there would also be that image issue they need to be aware (I haven't included the images yet).

It is done this way since I expect people only to use the mixins and variables but not other stuff at the moment. If it appears there is the need for full control over styling, which means the pre-built css isn't necessary any more, then I'd add support for excluding that file, similar to how the theme support is implemented.

**Comment by ramijarrar** [comment]
Why not just separate the js and provide different code for Sass/pre-built CSS users? A lot of the current complexity would probably be eliminated that way and the separation of concerns is already viisble (for example, Sass users are not interested in color settings/theming code while CSS users don't have any issues with image paths).

**Comment by Zodiase** [comment]
@ramijarrar I have thought about the option of having `mdl` as the js package and maybe `mdl-theme` as the pre-built css package and maybe `mdl-scss` for the scss files. But that way there'd be more packages I have to maintain and for the users there would be at least two copies of mdl distribution package with each only serving half the purpose. More importantly, does that also mean if they don't want to use the default theme, they have to go with scss?

**Comment by ramijarrar** [comment]
There are other ways to accomplish this if you don't want to separate the package (on the same wavelength as the settings file you implemented) it's just that the added complexity will probably mean things will start getting out of hand really quickly. 

Now with regards to color settings, I don't see any reason why the original code you wrote would no longer be valid?

**Comment by ramijarrar** [comment]
There are other ways to accomplish this if you don't want to separate the package (on the same wavelength as the settings file you implemented) it's just that the added complexity will probably mean things will start getting out of hand really quickly. 

Now with regards to color settings, I don't see any reason why the original code you wrote would no longer be valid?

**Comment by Zodiase** [comment]
@ramijarrar I'd like to know if there are other ways of making a package configurable. I barely came up with that solution and it's really hacky and isn't friendly at all (with the necessity to reset an app just to change the theme).

If I allow the theme package to be configurable while having a separate scss package, I don't see why it's worse having everything together. By default the scss files do nothing. If people want to use any of them, go ahead. If they want to go all the way with scss, just put the settings file there to disable the default theme then. Straight forward to me.

**Comment by ramijarrar** [comment]
As long as there is a way to disable the default theme it should be fine.

**Comment by ramijarrar** [comment]
The other thing is you could just let the user bring his own css/Sass and forget the whole settings which is really hacky as you said and adds a lot of complexity.

**Comment by Zodiase** [comment]
@ramijarrar I somehow believe to most meteor users maintaining the correct version of css file corresponding to a js package isn't fun. :p

**Comment by Zodiase** [comment]
@ramijarrar I just added support for disabling theme by having `"theme": false` in the settings file.

**Comment by ramijarrar** [comment]
Great, I'll let you know if I have any problems with it.

Btw - are you still planning to fix the images issue for this release?

**Comment by pgrm** [comment]
Hi @Zodiase, great job with this package!

I'm using the `add_scss_files` branch and it works great. except for 1 thing:

I've disabled the theme and also can see in the console:

```
MDL settings found. /Users/peter/Sync/GitHub/dict_builder/zodiase-mdl.json
MDL settings loaded { theme: false }
MDL Theme disabled.
```

I've added a custom variables file to define a theme (and later on overwrite other default values) with:

```
@import '{zodiase:mdl}/src/color-definitions';

$trim-color-classes: true;

// Use color primarily for emphasis. Choose colors that fit with
// your brand and provide good contrast between visual components.
$color-primary: $palette-deep-orange-500;
$color-primary-dark: $palette-deep-orange-700;
$color-accent: $palette-light-green-A200;

// Our primary is dark, so use $color-dark-contrast for overlaid text.
$color-primary-contrast: $color-dark-contrast;
// Our accent is light, so use $color-light-contrast for overlaid text.
$color-accent-contrast: $color-light-contrast;

@import '{zodiase:mdl}/src/variables';
```

and in my main scss file, I'm adding the following at the top:

```
@import 'variables';
@import '{zodiase:mdl}/src/material-design-lite';
```

everything works great, except that now I have mdl defined twice. Once with the default colors (in the beginning) and once with my theme.

Once the css code will get minified, a good minifier should be able to remove the first definition of everything and only keep the useful code, so it shouldn't be too much problem, but any idea how this can be solved anyway?

**Update**

just wanted to confirm it and turns out, they stay together also minified. Maybe there is any way the minifier could be tweaked? Or just give the option in the plugin, not to automatically include the scss files. Otherwise, I'll just use https://cssminifier.com/ before production. That service could remove my unused css easily.

thx

**Comment by Zodiase** [comment]
@pgrm I tested and don't see the css code appearing twice. Here's my `.scss` file:

``` SCSS
// Load color definitions to use.
@import '{zodiase:mdl}/src/color-definitions';

$trim-color-classes: true;

// Use color primarily for emphasis. Choose colors that fit with
// your brand and provide good contrast between visual components.
$color-primary: $palette-deep-orange-500;
$color-primary-dark: $palette-deep-orange-700;
$color-accent: $palette-light-green-A200;

// Our primary is dark, so use $color-dark-contrast for overlaid text.
$color-primary-contrast: $color-dark-contrast;
// Our accent is light, so use $color-light-contrast for overlaid text.
$color-accent-contrast: $color-light-contrast;

// Load mdl.
@import '{zodiase:mdl}/src/material-design-lite';
```

Note that different from your code clips, I don't need to import mdl variables in my variables portion (although I didn't split the code into two separate files).

Maybe I'm understanding your issue wrong. If so could you elaborate?

Update:
Check out the newly added test case 7b622d3f194e49a50e03e302a2027411f3ce5e99.

**Comment by Zodiase** [comment]
@ramijarrar I think adding <del>`$image_path: '{zodiase:mdl}/src/images'`</del>`$image_path: "packages/zodiase_mdl/src/images";` to a new scss file say `{zodiase:mdl}/theme.scss` and let people load that would work just fine? Of course I also need to actually add the images to the assets.

Is there a way the user might change `approot/packages` to a different address?

**Comment by pgrm** [comment]
@Zodiase sry, my fault - you're absolutely right!

There was still the old css file in the compiled meteor folder which has been added. After running

```
meteor reset
```

everything works now, thx :+1: also with having multiple files, ...

**Comment by Zodiase** [comment]
Release `1.0.6_3` is out. You may get rid of those local packages. :)

**Comment by ramijarrar** [comment]
@Zodiase The implementation for the images looks good to me (https://github.com/Zodiase/meteor-mdl/commit/07385bacf317a773dc41555ea3bf58545889574a) and no, it's not possible to change `approot/packages` :)

Thanks for all your work on this.

**Comment by Zodiase** [comment]
Hi guys I'm sorry if any of you experienced issue #26 when using the package from atmosphere and trying to use the settings file. The issue should be fixed in the latest release `1.0.6_5`. Unfortunately the fix also broke backward compatibility mostly related to settings and SASS files. Please refer to the README file for updates.
